subprojects/build-operations/src/main/java/org/gradle/trace/buildops/BuildOperationTrace.java

```diff
@@ -35,24 +34,18 @@
 
     private final BuildEventListenerRegistryInternal registry;
     private final BuildServiceRegistry sharedServices;
-    private final GradleInternal gradle;
 
     public BuildOperationTrace(GradleInternal gradle) {
-        this.gradle = gradle;
         this.registry = gradle.getServices().get(BuildEventListenerRegistryInternal.class);
         this.sharedServices = gradle.getSharedServices();
     }
 
     public BuildOperationTrace measureGarbageCollection(File outFile) {
```

Comment:
I wonder if this whole method should go to a new class. It is not really related to the whole `BuildOperationTrace` business. WDYT?

Reply:
That makes sense, though we probably should move the configuration time measurement, too, as it is not really about build operation tracing either. Let's do that in a separate PR, so we can merge this now and get going with collecting GC data in Gradle.